### PR TITLE
Adding Wireless hacking doc

### PR DESCRIPTION
**Pull request by rvs**
Signed-off-by: Roman Shaposhnik <rvs@zededa.com>

### CODE REVIEW

**Comment by rvs** [comment]
Yo @deitch @eriknordmark -- any chance you guys can take a look? I'll trade you this one for @eriknordmark review ;-)

File: docs/WIRELESS.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# Support for wireless connectivity in EVE
+
+Given that the biggest use case for EVE is in Edge Computing and IoT it is no surprise that wireless connectivity plays a major role in connecting edge applications. Currently EVE supports 2 kinds of wireless networking devices: WiFi cards and Cellular GSM Modems. When it comes to utilizing either of these EVE relies on its controller to supply two critical pieces of information:
+
+* Name of the wireless network to be joined ([SSID](https://en.wikipedia.org/wiki/Service_set_(802.11_network)#SSID) for WiFi and [APN](https://en.wikipedia.org/wiki/Access_Point_Name) for Cellular).
+* Pre-shared secret required to joining the network ([WPA/WPA2/WEP](https://en.wikipedia.org/wiki/Wi-Fi_Protected_Access) and [username/password](https://www.digi.com/resources/documentation/Digidocs/90001399-13/tasks/t-set-up-and-configure-mobile-connectivity.htm) for Cellular).
+
+This dependency on a controller creates a bit of a chicken-and-an-egg problem when the wireless connectivity itself happens to be the only means of reaching the controller. In those cases EVE uses pre-loaded controller configuration to bootstrap the entire process.
```
Comment (deitch) [inline docs/WIRELESS.md:8]: Really glad to see this raised up-front. How does the device know if it should be going down wifi vs cellular vs wired, and if it needs any extra credentials to connect to the network?

File: docs/WIRELESS.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+The rest of this document will be focused on details of wireless support in EVE. It must be noted that currently we're relying on a patchwork of different open source projects to do that, but we're also looking at things like [oFono](https://en.wikipedia.org/wiki/OFono) and [Android's RIL](https://en.wikipedia.org/wiki/Radio_Interface_Layer) for a more hollistic support.
+
+## WiFi
+
+In general, WiFi support in EVE is pretty straightforward and largely depends on enabling required driver in the Linux kernel and finding an appropriate firmware binary blob to be loaded by the device driver. Please refer to our [new hardware bringup](HARDWARE-BRINGUP.md) document for more details on the former and make sure to checkout our [firmware package](../pkg/firmware) for the later.
```
Comment (deitch) [inline docs/WIRELESS.md:14]: grammatical: "enabling the required driver"

Comment (deitch) [reply docs/WIRELESS.md:14]: typo: "for the latter"

File: docs/WIRELESS.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+## Cellular GSM Modems
+
+Compared to WiFi, cellular GSM modems present a major challenge. This is partially understandable, since radio regulations in different countries tend to be partially incompatible and at times [simply insane](https://www.afcea.org/content/disruptive-design-have-you-seen-flurry-acts-5g-security). All of this conspires to create a technology landscapte where modem manufacturers are forced to produce highly capable and highly configurable units that take onto very different personalities when loaded with firmware and given their final configuration.
```
Comment (deitch) [inline docs/WIRELESS.md:18]: "simple insane" - I love this!

File: docs/WIRELESS.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+As with any hardware, you will typically go through the following steps:
+
+1. making sure that the modem device is connected to the host edge node through one of the common busses (this doesn't guarantee that the host can communicate with the device, but at least you should see it registered on the bus)
+2. making sure that the modem is configured in such a way that you can actually communicate with it
+3. making sure that you can drive the modem and instruct it to connect to the wireless provider
+4. making sure that a modem connected to the wireless provider can establish a data connection and reflect it back to the host as a network interface (typically `wwanX`)
+
+#### 1. Basic bus connectivity
+
+Alsmot all well known GSM modems get hooked to a USB bus (most of them communicate using USB2 but some try USB3 and then [trouble may ensue](https://forum.sierrawireless.com/t/mc7455-not-recognized-at-boot/9735) and you may need to [do soldering](https://forum.sierrawireless.com/t/effect-of-removing-usb-3-0-pins-of-mc7455/12702/2)). From there, a single device will expose different interfaces to accommodate various ways of interacting with it. These include:
```
Comment (deitch) [inline docs/WIRELESS.md:35]: Typo "Almost"

File: docs/WIRELESS.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+* traditional serial interface capable of accepting AT-commands (popular choices of drivers include [Qualcom's qcserial](https://github.com/torvalds/linux/blob/master/drivers/usb/serial/qcserial.c) and [Sierra Wireless serial](https://github.com/torvalds/linux/blob/master/drivers/usb/serial/sierra.c)). This interface is NOT capable of directly transmitting Ethernet frames and requires an old-school PPP support
```
Comment (deitch) [inline docs/WIRELESS.md:37]: Typo: "Qualcomm"

File: docs/WIRELESS.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+* traditional serial interface used for special purposes such as [GPS NMEA port](https://en.wikipedia.org/wiki/NMEA_0183) or [Diagnostic Monitoring](https://forum.sierrawireless.com/t/sierra-linux-qmi-dmcapture-sh-not-working/6373)
+* One of the CDC (no, not [that CDC](https://www.cdc.gov/) - rather [communications device class](https://en.wikipedia.org/wiki/USB_communications_device_class)) interfaces capable of direct transport of Ethernet frames:
+  * [CDC MBIM](https://www.kernel.org/doc/Documentation/networking/cdc_mbim.txt)
+  * [CDC QMI](https://en.wikipedia.org/wiki/Qualcomm_MSM_Interface)
+
+To put it all together, a single GSM modem is likely to look something like this when inspected by the `lsusb -t` command:
+
+```bash
```
Comment (deitch) [inline docs/WIRELESS.md:45]: Nit: this might be better formatted as "console" rather than "bash"

File: docs/WIRELESS.md (hunk 1 above)
Comment (eriknordmark) [reply docs/WIRELESS.md:8]: Makes sense to add a reference to ./CONFIG.md which describes that json file.
But we must also request @naiming-zededa to add a WiFi and WLAN example to CONFIG.md.

File: docs/WIRELESS.md (hunk 3 above)
Comment (eriknordmark) [reply docs/WIRELESS.md:18]: Typo: landscapte
Might make sense to spell check the document

File: docs/WIRELESS.md (hunk 2 above)
Comment (eriknordmark) [reply docs/WIRELESS.md:14]: When I click on the new hardware bringup link I get nothing. Is this a new page? Or does it need to be a relative ./HARDWARE-BRINGUP.md URL?

File: docs/WIRELESS.md (hunk 1 above)
Comment (eriknordmark) [reply docs/WIRELESS.md:8]: @rvs did you see this and my other comments about missing references to HARDWARE-BRINGUP and FIRMWARE.md?

Comment (rvs) [reply docs/WIRELESS.md:8]: I've been trying to write a treatise @nordmark ;-) However, you're right an MVP now is better than treatise tomorrow. Please take a look at the MVP of that doc that's available in the updated PR.

File: docs/WIRELESS.md (hunk 2 above)
Comment (rvs) [reply docs/WIRELESS.md]: Fixed now @eriknordmark

File: docs/WIRELESS.md (hunk 3 above)
Comment (rvs) [reply docs/WIRELESS.md]: I did ;-) but it was an ispell(1) so.... ;-)

File: docs/WIRELESS.md (hunk 4 above)
Comment (rvs) [reply docs/WIRELESS.md]: fixed

File: docs/WIRELESS.md (hunk 5 above)
Comment (rvs) [reply docs/WIRELESS.md]: fixed

File: docs/WIRELESS.md (hunk 6 above)
Comment (rvs) [reply docs/WIRELESS.md]: fixed

**Comment by rvs** [comment]
Hey @deitch @eriknordmark -- I've fixed the nits and update the PR -- will merge tomorrow unless somebody hollers.

**Pull request merged**